### PR TITLE
Create an account on node first boot by specifying a password

### PR DESCRIPTION
**Pull request by jleeh**
When I've been trying to create auto-scaling CL nodes on AWS I've hit the issue that the node won't boot because it's trying to get stdin which causes it to throw a fatal.

Without setting up an account on boot using the PASSWORD env variable, I couldn't get it to boot inside a container in daemon mode.

Feel free to rip into this PR 😄

### CODE REVIEW

**Comment by jleeh** [comment]
The build for this failed but due to using the smartcontractkit/chainlink internal packages, so I can't show proof of the unit tests working on a forked PR:
https://travis-ci.org/LinkPoolio/chainlink/jobs/345376402

**Comment by thodges-gh** [comment]
It seems that this tries to automatically apply a password if one was already set up where master will still allow the user to enter one.

**Comment by thodges-gh** [comment]
From a security standpoint, I'd rather shy away from having the `PASSWORD` environment variable share the same value as the password used to unlock a keystore.

The `PASSWORD` environment variable is only used for authenticating with the node via the API. The password that is prompted when running a node is to unlock the keystore which holds currency.

**Comment by jleeh** [comment]
Cheers @thodges-gh, didn't realise the difference in passwords. I'll update it so it doesn't include `EnvVar` in main.go, so you need to explicitly pass it in.

**Comment by thodges-gh** [comment]
Cool, this fixes the auto password.

**Comment by thodges-gh** [comment]
Okay, interesting stuff.

So for documenting what this change does, currently, if you **first** invoke the node with something like `go run main.go n -p test" it will return the following output:
```
Cannot authenticate with password because there are no accounts
```
With this change, and using the same method to invoke the node for the first time (meaning there are no keys in the `~/.chainlink/keys` directory), the node would create a key with the given password and return the following output:
```
There are no accounts, creating a new account with the specified password
```

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ tmp/
 internal/devnet/db.bolt
 internal/devnet/log.jsonl
 /chainlink
+.idea/
```
Comment (se3000) [inline .gitignore]: This seems specific to your dev environment. Could you move it to your global gitignore?

**Comment by jleeh** [comment]
Updated.

File: cmd/authenticator_test.go [hunk 2]
```diff
@@ -44,8 +44,9 @@ func TestTerminalAuthenticatorWithNoAcctWithInitialPwd(t *testing.T) {
 	}}
 
 	auth.Authenticate(app.Store, "somepassword")
-	assert.Equal(t, true, exited)
-	assert.Equal(t, 1, rval)
```
Comment (se3000) [inline cmd/authenticator_test.go]: You can get rid of the `rval` variable in this test too, looks like it's not used anymore.

Comment (jleeh) [reply cmd/authenticator_test.go]: Removed. Sorry about slipping with this, my excuse was that it was AM.

Comment (se3000) [reply cmd/authenticator_test.go]: Haha, no worries. Thanks for fixing it up!

**Pull request merged**